tchannel/thrift/rw.py

```diff
@@ -0,0 +1,369 @@
+        @tornado.gen.coroutine
+        def submitOrder(request):
+            args = request.body
+
+            if args.coffee:
+                yield tchannel.thrift(
+                    coffee.CoffeeService.order(args.coffee)
+                )
+
+            # ...
+
+    The returned module contains, one top-level type for each struct, enum,
+    union, exeption, and service defined in the Thrift file. For each service,
+    the corresponding class contains a classmethod for each function defined
+    in that service that accepts the arguments for that function and returns a
+    ``ThriftRequest`` capable of being sent via ``TChannel.thrift``.
+
+    Note that the ``path`` accepted by ``load`` must be either an absolute
+    path or a path relative to the *the current directory*. If you need to
+    refer to Thrift files relative to the Python module in which ``load`` was
+    called, use the ``__file__`` magic variable.
+
+    .. code-block:: python
+
+        # Given,
+        #
+        #   foo/
+        #     myservice.thrift
+        #     bar/
+        #       x.py
+        #
+        # Inside foo/bar/x.py,
+
+        path = os.path.join(
+            os.path.dirname(__file__), '../myservice.thrift'
+        )
+
+    The returned value is a valid Python module. You can install the module by
+    adding it to the ``sys.modules`` dictionary. This will allow importing
+    items from this module directly. You can use the ``__name__`` magic
+    variable to make the generated module a submodule of the current module.
+    For example,
+
+    .. code-block:: python
+
+        # foo/bar.py
+
+        import sys
+        from tchannel import thrift
+
+        donuts = = thrift.load('donuts', 'donuts.thrift')
+        sys.modules[__name__ + '.donuts'] = donuts
+
+    This installs the module generated for ``donuts.thrift`` as the module
+    ``foo.bar.donuts``. Callers can then import items from that module
+    directly. For example,
+
+    .. code-block:: python
+
+        # foo/baz.py
+
+        from foo.bar.donuts import DonutsService, Order
+
+        def baz():
+            return tchannel.thrift(
+                DonutsService.submitOrder(Order(..))
+            )
+
+    :param str service:
+        Name of the service that the Thrift file represents. This name will be
+        used to route requests through Hyperbahn.
+    :param str path:
+        Path to the Thrift file. If this is a relative path, it must be
+        relative to the current directory.
+    :param str hostport:
+        Clients can use this to specify the hostport at which the service can
+        be found. If omitted, TChannel will route the requests through known
+        peers. This value is ignored by servers.
+    :param str module_name:
+        Name used for the generated Python module. Defaults to the name of the
+        Thrift file.
+    """
+    assert service, 'service is required'
+    assert path, 'path is required'
```

Comment:
probably shouldn't have assert in public api, but since this is on startup this would be to developers only... maybe just add a TODO